# **Section 1: Dataframe Setup** #

### Data Preprocessing ###


Data sourced from https://knowthechain.org/benchmark/, https://knowthechain.org/wp-content/uploads/KTC-2022-ICT-Benchmark-Report.pdf

**2022**

Excel file modified for Pandas CSV readable format as follows:

- Columns 'Total Benchmark Score' and '2022 Rank' moved from 51 and 52 to 2 and 3
- Moved 'Year of Inclusion' column from D to F (aka moving to the left the  columns 'Country', 'Region', 'Market Cap ($1bln)')
- Consolidated and moved column headers on rows 2-4 to row 1
- Appended column titles for rows BA-BG to include Theme number (e.g. 'Monitoring' to 'Theme 5: Monitoring') 
- Rows BA to BG (Averaage total benchmark score disaggregated by Theme) moved to H-N
- Sorted row data by column B ('Total benchmark score') in descending order


**2020**

Excel file modified for Pandas CSV readable format as follows:

- Consolidated and moved column headers on rows 2-4 to row 1
- Column A 'Company (full analysis companies only)' renamed to 'Company Name'
- Column 'Total' moved from AH to B and renamed 'Total benchmark score' 
- Column AI 'Rank 2020' moved from AI to C and renamed '2020 Rank'
- Column D 'Market Cap in US$ bn' moved from D to F and renamed 'Market cap (US$bn)'
- New columns added H-N to reflect total scores for each theme (columns T, W, AA, AF, AJ, AM, AP) in same format as 2022 csv data file
- Sorted row data by column B ('Total benchmark score') in descending order 

**2018**

Excel file modified for Pandas CSV readable format as follows:

- Column E 'Benchmark Score 2018' moved to column B and renamed 'Total benchmark score'
- Column F 'Benchmark rank 2018' moved to column C and renamed '2018 rank'
- Column D 'Market Cap in US$ billion' moved to column F and renamed 'Market cap (US$bn)
- Column F '2016 company' string data replaced with int data 'Year of inclusion' to reflect criteria used in subsequent 2020 and 2022 datasets
- New column addedx in column E titled 'Region' to reflect criteria included in subsequent 2020 and 2022 datasets
- New columns added H-N to reflect total scores for each theme (columns T, W, AA, AF, AJ, AM, AP) in same format as 2022 csv data file
- Sorted row data by column B ('Total benchmark score') in descending order 

### Basic Dataframe Creation ###

***Process for calling data in Juypter notebook***

In [92]:
import pandas as pd
import plotly.express as px
import numpy as np
import plotly.graph_objects as go

def process_csv(file_path):
    df = pd.read_csv(file_path)
    return df

ktc_ict_2022 = process_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")
display(ktc_ict_2022.iloc[:, :7].head(100))

ktc_ict_2020 = process_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2020 KCT ICT.csv")
display(ktc_ict_2020.iloc[:, :7].head(100))

ktc_ict_2018 = process_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2018 KCT ICT.csv")
display(ktc_ict_2018.iloc[:, :7].head(100))


Company Name  Total benchmark score  \
0             Hewlett Packard Enterprise Co. (HPE)              63.420455   
1                                      Intel Corp.              60.238636   
2                               Cisco Systems Inc.              54.647727   
3                                       Apple Inc.              51.806818   
4                                          HP Inc.              46.147727   
..                                             ...                    ...   
55                                   Kyocera Corp.               2.590909   
56      Sunny Optical Technology (Group) Co., Ltd.               1.363636   
57                   BOE Technology Group Co. Ltd.               0.000000   
58  Hangzhou Hikvision Digital Technology Co. Ltd.               0.000000   
59                NAURA Technology Group Co., Ltd.               0.000000   

    2022 Rank        Country         Region  Market cap (US$bn)  \
0           1  United States  North America            18770.52   
1           2  United States  North America           200096.40   
2           3  United States  North America           231293.54   
3           4  United States  North America          2711977.42   
4           5  United States  North America            40660.86   
..        ...            ...            ...                 ...   
55         55          Japan           Asia            21434.67   
56         57          China           Asia            33158.48   
57         58          China           Asia            28707.64   
58         58          China           Asia            72120.00   
59         58          China           Asia            31838.53   

    Year of inclusion  
0                2018  
1                2016  
2                2016  
3                2016  
4                2016  
..                ...  
55               2018  
56               2022  
57               2016  
58               2020  
59               2022  

[60 rows x 7 columns]

Company Name  Total benchmark score  \
0             Hewlett Packard Enterprise Co. (HPE)              69.944489   
1                                          HP Inc.              69.214093   
2                     Samsung Electronics Co. Ltd.              69.123825   
3                                      Intel Corp.              67.933587   
4                                       Apple Inc.              67.515003   
..                                             ...                    ...   
44                                   Keyence Corp.               5.612623   
45                   BOE Technology Group Co. Ltd.               4.570914   
46  Hangzhou Hikvision Digital Technology Co. Ltd.               2.860572   
47                       Largan Precision Co. Ltd.               2.860572   
48                                    Xiaomi Corp.               0.000000   

    2020 Rank        Country         Region  Market cap (US$bn)  \
0           1  United States  North America            26.13972   
1           2  United States  North America            38.36673   
2           2    South Korea           Asia           309.60738   
3           4  United States  North America           225.29520   
4           4  United States  North America           851.72579   
..        ...            ...            ...                 ...   
44         45          Japan           Asia            73.72563   
45         46          China           Asia            33.37354   
46         47          China           Asia            58.56661   
47         47         Taiwan           Asia            18.41915   
48         49          China           Asia            37.43900   

    Year of inclusion  
0                2018  
1                2016  
2                2016  
3                2016  
4                2016  
..                ...  
44               2016  
45               2016  
46               2020  
47               2018  
48               2020  

[49 rows x 7 columns]

Company Name  Total benchmark score  2018 rank  \
0                    Intel Corporation                     75          1   
1                              HP Inc.                     72          2   
2   Hewlett Packard Enterprise Company                     71          3   
3                           Apple Inc.                     71          3   
4                NXP Semiconductors NV                     63          5   
..                                 ...                    ...        ...   
35   Microchip Technology Incorporated                      7         35   
36                Corning Incorporated                      6         37   
37                       Broadcom Inc.                      6         37   
38       BOE Technology Group Co. Ltd.                      4         39   
39          Largan Precision Co., Ltd.                      0         40   

          Country         Region  Market cap (US$bn)  Year of inclusion  
0   United States  North America                 225               2016  
1   United States  North America                  38               2016  
2   United States  North America                  26               2018  
3   United States  North America                 852               2016  
4     Netherlands         Europe                  41               2018  
..            ...            ...                 ...                ...  
35  United States  North America                  22               2018  
36  United States  North America                  27               2018  
37  United States  North America                 102               2016  
38          China           Asia                  33               2016  
39         Taiwan           Asia                  18               2018  

[40 rows x 7 columns]

### Basic Data Information ###

***Column Headers***

In [115]:
ktc_ict_2022.iloc[:, :14].columns

Index(['Company Name', 'Total benchmark score', '2022 Rank', 'Country',
       'Region', 'Market cap (US$bn)', 'Year of inclusion',
       'Theme 1: Commitment & Governance',
       'Theme 2: Traceability & Risk Assessment',
       'Theme 3: Purchasing Practices', 'Theme 4: Recruitment',
       'Theme 5: Worker Voice', 'Theme 6: Monitoring', 'Theme 7: Remedy'],
      dtype='object')

In [114]:
ktc_ict_2020.iloc[:, :14].columns

Index(['Company Name', 'Total benchmark score', '2020 Rank', 'Country',
       'Region', 'Market cap (US$bn)', 'Year of inclusion',
       'Theme 1: Commitment & Governance',
       'Theme 2: Traceability & Risk Assessment',
       'Theme 3: Purchasing Practices', 'Theme 4: Recruitment',
       'Theme 5: Worker Voice', 'Theme 6: Monitoring', 'Theme 7: Remedy'],
      dtype='object')

In [116]:
ktc_ict_2018.iloc[:, :14].columns

Index(['Company Name', 'Total benchmark score', '2018 rank', 'Country',
       'Region', 'Market cap (US$bn)', 'Year of inclusion',
       'Theme 1: Commitment & Governance',
       'Theme 2: Traceability & Risk Assessment',
       'Theme 3: Purchasing Practices', 'Theme 4: Recruitment',
       'Theme 5: Worker Voice', 'Theme 6: Monitoring', 'Theme 7: Remedy'],
      dtype='object')

***Data Types***

In [118]:
ktc_ict_2022.iloc[:, :14].dtypes

Company Name                      object
Total benchmark score            float64
2022 Rank                          int64
Country                           object
Region                            object
                                  ...   
Theme 3: Purchasing Practices    float64
Theme 4: Recruitment             float64
Theme 5: Worker Voice            float64
Theme 6: Monitoring                int64
Theme 7: Remedy                  float64
Length: 14, dtype: object

In [119]:
ktc_ict_2020.iloc[:, :14].dtypes

Company Name                      object
Total benchmark score            float64
2020 Rank                          int64
Country                           object
Region                            object
                                  ...   
Theme 3: Purchasing Practices    float64
Theme 4: Recruitment             float64
Theme 5: Worker Voice            float64
Theme 6: Monitoring                int64
Theme 7: Remedy                  float64
Length: 14, dtype: object

In [120]:
ktc_ict_2018.iloc[:, :14].dtypes

Company Name                     object
Total benchmark score             int64
2018 rank                         int64
Country                          object
Region                           object
                                  ...  
Theme 3: Purchasing Practices     int64
Theme 4: Recruitment              int64
Theme 5: Worker Voice             int64
Theme 6: Monitoring               int64
Theme 7: Remedy                   int64
Length: 14, dtype: object

# **Section 2: Basic Content Information** # 

### Geographic Content ###

***Regions***

Three regions (North America, Europe, Asia) represented in all three benchmarks (2018, 2020, 2022)

In [127]:
ktc_ict_2022['Region'].unique()

array(['North America', 'Asia', 'Europe'], dtype=object)

***Countries***

Ireland was added in the 2022 benchmark (Seagate PLC)

In [126]:
ktc_ict_2018['Country'].unique()

array(['United States', 'Netherlands', 'South Korea', 'Taiwan', 'Sweden',
       'Japan', 'Finland', 'Switzerland', 'Germany', 'China'],
      dtype=object)

In [125]:
ktc_ict_2020['Country'].unique()

array(['United States', 'South Korea', 'Netherlands', 'Sweden', 'Finland',
       'Japan', 'Taiwan', 'Switzerland', 'Germany', 'China'], dtype=object)

In [123]:
ktc_ict_2022['Country'].unique()

array(['United States', 'South Korea', 'Sweden', 'Netherlands',
       'Switzerland', 'Ireland', 'Japan', 'Finland', 'Taiwan', 'Germany',
       'China'], dtype=object)

### Company Information ###

***Companies assessed in 2016***

In [128]:
companies_2016 = ktc_ict_2022[ktc_ict_2022['Year of inclusion'] == 2016]['Company Name'].unique()
print(companies_2016)

['Intel Corp.' 'Cisco Systems Inc.' 'Apple Inc.' 'HP Inc.'
 'Samsung Electronics Co. Ltd.' 'Telefonaktiebolaget LM Ericsson (publ)'
 'Microsoft Corp.' 'Texas Instruments Inc.' 'Qualcomm Inc.'
 'Broadcom Inc.' 'Murata Manufacturing Co. Ltd.' 'Canon Inc.'
 'Taiwan Semiconductor Manufacturing Co. Ltd.' 'Hitachi Ltd.'
 'SK Hynix Inc.' 'Keyence Corp.' 'ASML Holding NV'
 'Hon Hai Precision Industry Co. Ltd. (Foxconn)'
 'BOE Technology Group Co. Ltd.']


***Companies assessed in 2018***

New Companies

In [137]:
companies_2018_new = ktc_ict_2022[ktc_ict_2022['Year of inclusion'] == 2018]['Company Name'].unique()
print(companies_2018_new)

['Hewlett Packard Enterprise Co. (HPE)' 'Amazon.com Inc.' 'Corning Inc.'
 'NXP Semiconductors NV' 'Western Digital Corp.' 'Lam Research Corp.'
 'Nokia Oyj' 'Micron Technology Inc.' 'Skyworks Solutions Inc.'
 'NVIDIA Corp.' 'Tokyo Electron Ltd.' 'Microchip Technology Inc.'
 'Analog Devices Inc.' 'Amphenol Corp.' 'Applied Materials Inc.'
 'TE Connectivity Ltd.' 'Infineon Technologies AG' 'Nintendo Co. Ltd.'
 'Kyocera Corp.']


All Companies

In [133]:
companies_2018 = ktc_ict_2022[(ktc_ict_2022['Year of inclusion'] == 2018) | (ktc_ict_2022['Year of inclusion'] == 2016)]['Company Name'].unique()
print(companies_2018)

['Hewlett Packard Enterprise Co. (HPE)' 'Intel Corp.' 'Cisco Systems Inc.'
 'Apple Inc.' 'HP Inc.' 'Samsung Electronics Co. Ltd.'
 'Telefonaktiebolaget LM Ericsson (publ)' 'Amazon.com Inc.' 'Corning Inc.'
 'NXP Semiconductors NV' 'Western Digital Corp.' 'Microsoft Corp.'
 'Lam Research Corp.' 'Nokia Oyj' 'Micron Technology Inc.'
 'Texas Instruments Inc.' 'Qualcomm Inc.' 'Skyworks Solutions Inc.'
 'NVIDIA Corp.' 'Broadcom Inc.' 'Tokyo Electron Ltd.'
 'Microchip Technology Inc.' 'Murata Manufacturing Co. Ltd.'
 'Analog Devices Inc.' 'Amphenol Corp.' 'Canon Inc.'
 'Applied Materials Inc.' 'Taiwan Semiconductor Manufacturing Co. Ltd.'
 'Hitachi Ltd.' 'TE Connectivity Ltd.' 'Infineon Technologies AG'
 'Nintendo Co. Ltd.' 'SK Hynix Inc.' 'Keyence Corp.' 'ASML Holding NV'
 'Hon Hai Precision Industry Co. Ltd. (Foxconn)' 'Kyocera Corp.'
 'BOE Technology Group Co. Ltd.']


***Companies assessed in 2020***

New Companies

In [139]:
companies_2020_new = ktc_ict_2022[ktc_ict_2022['Year of inclusion'] == 2020]['Company Name'].unique()
print(companies_2020_new)

['Best Buy Co. Inc.' 'STMicroelectronics NV' 'Dell Technologies Inc.'
 'Walmart Inc.' 'Sony Corp.' 'LG Electronics Inc.'
 'Motorola Solutions Inc.' 'KLA Corp.' 'Renesas Electronics Corp.'
 'Fujifilm Holdings Corp.' 'Panasonic Corp.' 'Hexagon AB' 'ZTE Corp.'
 'Xiaomi Corp.' 'Hangzhou Hikvision Digital Technology Co. Ltd.']


All Companies

In [136]:
companies_2020 = ktc_ict_2022[(ktc_ict_2022['Year of inclusion'] == 2020) | (ktc_ict_2022['Year of inclusion'] == 2018) | (ktc_ict_2022['Year of inclusion'] == 2016)]['Company Name'].unique()
print(companies_2020)

['Hewlett Packard Enterprise Co. (HPE)' 'Intel Corp.' 'Cisco Systems Inc.'
 'Apple Inc.' 'HP Inc.' 'Samsung Electronics Co. Ltd.'
 'Telefonaktiebolaget LM Ericsson (publ)' 'Amazon.com Inc.' 'Corning Inc.'
 'NXP Semiconductors NV' 'Western Digital Corp.' 'Best Buy Co. Inc.'
 'Microsoft Corp.' 'STMicroelectronics NV' 'Dell Technologies Inc.'
 'Walmart Inc.' 'Lam Research Corp.' 'Sony Corp.' 'Nokia Oyj'
 'Micron Technology Inc.' 'Texas Instruments Inc.' 'Qualcomm Inc.'
 'Skyworks Solutions Inc.' 'NVIDIA Corp.' 'LG Electronics Inc.'
 'Broadcom Inc.' 'Tokyo Electron Ltd.' 'Microchip Technology Inc.'
 'Motorola Solutions Inc.' 'Murata Manufacturing Co. Ltd.'
 'Analog Devices Inc.' 'Amphenol Corp.' 'Canon Inc.'
 'Applied Materials Inc.' 'Taiwan Semiconductor Manufacturing Co. Ltd.'
 'Hitachi Ltd.' 'KLA Corp.' 'TE Connectivity Ltd.'
 'Infineon Technologies AG' 'Renesas Electronics Corp.'
 'Nintendo Co. Ltd.' 'SK Hynix Inc.' 'Fujifilm Holdings Corp.'
 'Panasonic Corp.' 'Keyence Corp.' 'Hexagon 

***Companies assessed in 2022***

New Companies

In [140]:
companies_2022_new = ktc_ict_2022[ktc_ict_2022['Year of inclusion'] == 2022]['Company Name'].unique()
print(companies_2022_new)

['Advanced Micro Devices Inc' 'Seagate Technology PLC'
 'Logitech International S.A.'
 'Semiconductor Manufacturing International Corp'
 'Luxshare Precision Industry Co. Ltd.'
 'Sunny Optical Technology (Group) Co., Ltd.'
 'NAURA Technology Group Co., Ltd.']


All Companies

In [141]:
companies_2022 = ktc_ict_2022[(ktc_ict_2022['Year of inclusion'] == 2022) | (ktc_ict_2022['Year of inclusion'] == 2020) | (ktc_ict_2022['Year of inclusion'] == 2018) | (ktc_ict_2022['Year of inclusion'] == 2016)]['Company Name'].unique()
print(companies_2022)

['Hewlett Packard Enterprise Co. (HPE)' 'Intel Corp.' 'Cisco Systems Inc.'
 'Apple Inc.' 'HP Inc.' 'Samsung Electronics Co. Ltd.'
 'Telefonaktiebolaget LM Ericsson (publ)' 'Amazon.com Inc.' 'Corning Inc.'
 'NXP Semiconductors NV' 'Western Digital Corp.' 'Best Buy Co. Inc.'
 'Microsoft Corp.' 'Advanced Micro Devices Inc' 'STMicroelectronics NV'
 'Seagate Technology PLC' 'Dell Technologies Inc.' 'Walmart Inc.'
 'Lam Research Corp.' 'Logitech International S.A.' 'Sony Corp.'
 'Nokia Oyj' 'Micron Technology Inc.' 'Texas Instruments Inc.'
 'Qualcomm Inc.' 'Skyworks Solutions Inc.' 'NVIDIA Corp.'
 'LG Electronics Inc.' 'Broadcom Inc.' 'Tokyo Electron Ltd.'
 'Microchip Technology Inc.' 'Motorola Solutions Inc.'
 'Murata Manufacturing Co. Ltd.' 'Analog Devices Inc.' 'Amphenol Corp.'
 'Canon Inc.' 'Applied Materials Inc.'
 'Taiwan Semiconductor Manufacturing Co. Ltd.' 'Hitachi Ltd.' 'KLA Corp.'
 'TE Connectivity Ltd.' 'Infineon Technologies AG'
 'Renesas Electronics Corp.' 'Nintendo Co. Ltd.' '

# **Section 3: Basic Descriptive Statistics** #

### Average, Median, and Standard Deviation of Benchmark Scores ###

***Average, Median, and Standard Deviation of Benchmark Scores for All Companies in 2018***

In [153]:
mean_2018 = ktc_ict_2018['Total benchmark score'].mean()
median_2018 = ktc_ict_2018['Total benchmark score'].median()
std_2018 = ktc_ict_2018['Total benchmark score'].std()
mean_region_2018 = ktc_ict_2018.groupby('Region')['Total benchmark score'].mean()
mean_region_2018 = mean_region_2018.sort_values(ascending=False)

print("Mean benchmark scores across all companies:", mean_2018)
print("Median benchmark scores across all companies:", median_2018)
print("Standard deviation of benchmark scores across all companies:", std_2018, "\n")
print(mean_region_2018)

Mean benchmark scores across all companies: 32.2
Median benchmark scores across all companies: 31.0
Standard deviation of benchmark scores across all companies: 21.543277849779976 

Region
North America    37.500000
Europe           35.666667
Asia             23.142857
Name: Total benchmark score, dtype: float64


***Average, Median, and Standard Deviation of Benchmark Scores for All Companies in 2020***

In [150]:
mean_2020 = ktc_ict_2020['Total benchmark score'].mean()
median_2020 = ktc_ict_2020['Total benchmark score'].median()
std_2020 = ktc_ict_2020['Total benchmark score'].std()
mean_region_2020 = ktc_ict_2020.groupby('Region')['Total benchmark score'].mean()
mean_region_2020 = mean_region_2020.sort_values(ascending=False)

print("Mean benchmark scores across all companies:", mean_2020)
print("Median benchmark scores across all companies:", median_2020)
print("Standard deviation of benchmark scores across all companies:", std_2020, "\n")
print(mean_region_2020)

Mean benchmark scores across all companies: 30.2848273732449
Median benchmark scores across all companies: 26.78185637
Standard deviation of benchmark scores across all companies: 20.85751177027481 

Region
North America    39.715280
Europe           30.281556
Asia             18.236258
Name: Total benchmark score, dtype: float64


***Average, Median, and Standard Deviation of Benchmark Scores for All Companies in 2022***

In [154]:
mean_2022 = ktc_ict_2022['Total benchmark score'].mean()
median_2022 = ktc_ict_2022['Total benchmark score'].median()
std_2022 = ktc_ict_2022['Total benchmark score'].std()
mean_region_2022 = ktc_ict_2022.groupby('Region')['Total benchmark score'].mean()
mean_region_2022 = mean_region_2022.sort_values(ascending=False)

print("Mean benchmark scores across all companies:", mean_2022)
print("Median benchmark scores across all companies:", median_2022)
print("Standard deviation of benchmark scores across all companies:", std_2022, "\n")
print(mean_region_2022)

Mean benchmark scores across all companies: 20.0840909091
Median benchmark scores across all companies: 14.045454544999998
Standard deviation of benchmark scores across all companies: 15.654284463757786 

Region
North America    28.866259
Europe           22.307955
Asia              9.643466
Name: Total benchmark score, dtype: float64


***Are the trend lines for these data affected by the inclusion of new companies during each year of the benchmark?***

In [157]:
# Filter dataframe to only include companies with Year of inclusion == 2018
ktc_ict_2022 = ktc_ict_2022[ktc_ict_2022['Year of inclusion'] == 2018]

mean_2022 = ktc_ict_2022['Total benchmark score'].mean()
median_2022 = ktc_ict_2022['Total benchmark score'].median()
std_2022 = ktc_ict_2022['Total benchmark score'].std()
mean_region_2022 = ktc_ict_2022.groupby('Region')['Total benchmark score'].mean()
mean_region_2022 = mean_region_2022.sort_values(ascending=False)

# Print results
print("Mean benchmark scores across all companies (Year of inclusion = 2018):", mean_2022)
print("Median benchmark scores across all companies (Year of inclusion = 2018):", median_2022)
print("Standard deviation of benchmark scores across all companies (Year of inclusion = 2018):", std_2022, "\n")
print(mean_region_2022)


Mean benchmark scores across all companies (Year of inclusion = 2018): 22.023325359526314
Median benchmark scores across all companies (Year of inclusion = 2018): 15.52272727
Standard deviation of benchmark scores across all companies (Year of inclusion = 2018): 15.03384347609765 

Region
North America    25.587121
Europe           21.014205
Asia              9.113636
Name: Total benchmark score, dtype: float64


***Top 5 and Bottom 5 in North Amercia***

In [42]:

ktc_ict_2022.loc[(ktc_ict_2022["Region"] == "North America")].sort_values(by='Total benchmark score', ascending=False).iloc[:, :6]

Company Name  Total benchmark score  2022 Rank  \
16  Hewlett Packard Enterprise Co. (HPE)              63.420455          1   
22                           Intel Corp.              60.238636          2   
11                    Cisco Systems Inc.              54.647727          3   
4                             Apple Inc.              51.806818          4   
20                               HP Inc.              46.147727          5   
..                                   ...                    ...        ...   
33               Motorola Solutions Inc.              13.500000         28   
3                    Analog Devices Inc.              12.818182         33   
2                         Amphenol Corp.              12.340909         35   
5                 Applied Materials Inc.              12.272727         35   
24                             KLA Corp.              12.000000         35   

          Country         Region  Market cap (US$bn)  
16  United States  North America            18770.52  
22  United States  North America           200096.40  
11  United States  North America           231293.54  
4   United States  North America          2711977.42  
20  United States  North America            40660.86  
..            ...            ...                 ...  
33  United States  North America            42761.47  
3   United States  North America            96868.35  
2   United States  North America            48189.08  
5   United States  North America           132902.05  
24  United States  North America            61881.55  

[26 rows x 6 columns]

***Top 5 and Bottom 5 in Europe***

In [43]:

ktc_ict_2022.loc[(ktc_ict_2022["Region"] == "Europe")].sort_values(by='Total benchmark score', ascending=False).iloc[:, :6]

Company Name  Total benchmark score  2022 Rank  \
53  Telefonaktiebolaget LM Ericsson (publ)              41.500000          7   
39                   NXP Semiconductors NV              39.465909         10   
49                   STMicroelectronics NV              28.988636         15   
44                  Seagate Technology PLC              27.420455         16   
28             Logitech International S.A.              23.659091         19   
37                               Nokia Oyj              22.431818         22   
52                    TE Connectivity Ltd.              11.113636         41   
21                Infineon Technologies AG              11.045455         41   
17                              Hexagon AB               8.863636         45   
6                          ASML Holding NV               8.590909         45   

        Country  Region  Market cap (US$bn)  
53       Sweden  Europe            33534.30  
39  Netherlands  Europe            59398.85  
49  Switzerland  Europe            44150.79  
44      Ireland  Europe            22858.03  
28  Switzerland  Europe            13218.97  
37      Finland  Europe            31270.20  
52  Switzerland  Europe            50229.41  
21      Germany  Europe            58680.49  
17       Sweden  Europe            39291.65  
6   Netherlands  Europe           326540.98

***Top 5 and Bottom 5 in Asia***

In [44]:

ktc_ict_2022.loc[(ktc_ict_2022["Region"] == "Asia")].sort_values(by='Total benchmark score', ascending=False).iloc[:, :6]

Company Name  Total benchmark score  \
43                    Samsung Electronics Co. Ltd.              45.522727   
48                                      Sony Corp.              23.170455   
27                             LG Electronics Inc.              14.454545   
55                             Tokyo Electron Ltd.              14.113636   
34                   Murata Manufacturing Co. Ltd.              12.954545   
..                                             ...                    ...   
25                                   Kyocera Corp.               2.590909   
50      Sunny Optical Technology (Group) Co., Ltd.               1.363636   
35                NAURA Technology Group Co., Ltd.               0.000000   
15  Hangzhou Hikvision Digital Technology Co. Ltd.               0.000000   
8                    BOE Technology Group Co. Ltd.               0.000000   

    2022 Rank      Country Region  Market cap (US$bn)  
43          5  South Korea   Asia           401477.49  
48         21        Japan   Asia           150653.89  
27         28  South Korea   Asia            15793.86  
55         28        Japan   Asia            81783.94  
34         33        Japan   Asia            47109.83  
..        ...          ...    ...                 ...  
25         55        Japan   Asia            21434.67  
50         57        China   Asia            33158.48  
35         58        China   Asia            31838.53  
15         58        China   Asia            72120.00  
8          58        China   Asia            28707.64  

[24 rows x 6 columns]

***Benchmark Scores Theme 1: Committment & Governance***

In [73]:
mean_scores_theme1 = ktc_ict_2022.groupby('Region')['Theme 1: Commitment & Governance'].mean()
mean_scores_theme1 = mean_scores_theme1.sort_values(ascending=False)
print(mean_scores_theme1)

Region
North America    56.410256
Europe           55.833333
Asia             26.562500
Name: Theme 1: Commitment & Governance, dtype: float64


***Benchmark Scores Theme 2: Traceability & Risk Assessment***

In [63]:
mean_scores_theme2 = ktc_ict_2022.groupby('Region')['Theme 2: Traceability & Risk Assessment'].mean()
mean_scores_theme2 = mean_scores_theme2.sort_values(ascending=False)
print(mean_scores_theme2)

Region
North America    37.126068
Europe           23.125000
Asia             10.619213
Name: Theme 2: Traceability & Risk Assessment, dtype: float64


***Benchmark Scores Theme 3: Purchasing Practices***

In [64]:
mean_scores_theme3 = ktc_ict_2022.groupby('Region')['Theme 3: Purchasing Practices'].mean()
mean_scores_theme3 = mean_scores_theme3.sort_values(ascending=False)
print(mean_scores_theme3)

Region
North America    4.487179
Asia             1.041667
Europe           0.000000
Name: Theme 3: Purchasing Practices, dtype: float64


***Benchmark Scores Theme 4: Recruitment***

In [65]:
mean_scores_theme4 = ktc_ict_2022.groupby('Region')['Theme 4: Recruitment'].mean()
mean_scores_theme4 = mean_scores_theme4.sort_values(ascending=False)
print(mean_scores_theme4)

Region
North America    25.673077
Europe           16.250000
Asia              6.979167
Name: Theme 4: Recruitment, dtype: float64


***Benchmark Scores Theme 5: Worker Voice***

In [66]:
mean_scores_theme5 = ktc_ict_2022.groupby('Region')['Theme 5: Worker Voice'].mean()
mean_scores_theme5 = mean_scores_theme5.sort_values(ascending=False)
print(mean_scores_theme5)

Region
North America    10.801282
Europe            6.250000
Asia              5.208333
Name: Theme 5: Worker Voice, dtype: float64


***Benchmark Scores Theme 6: Monitoring*** 

In [67]:
mean_scores_theme6 = ktc_ict_2022.groupby('Region')['Theme 6: Monitoring'].mean()
mean_scores_theme6 = mean_scores_theme6.sort_values(ascending=False)
print(mean_scores_theme6)

Region
North America    22.500000
Europe           18.500000
Asia              4.791667
Name: Theme 6: Monitoring, dtype: float64


***Benchmark Scores Theme 7: Remedy***

In [68]:
mean_scores_theme7 = ktc_ict_2022.groupby('Region')['Theme 7: Remedy'].mean()
mean_scores_theme7 = mean_scores_theme7.sort_values(ascending=False)
print(mean_scores_theme7)

Region
North America    23.461538
Europe           23.000000
Asia              2.187500
Name: Theme 7: Remedy, dtype: float64


# **Section 3: Data Visualisation** # 

## Basic Data Visualisation ##

In [20]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

# Load the three dataframes
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")
ktc_ict_2020 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2020 KCT ICT.csv")
ktc_ict_2018 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2018 KCT ICT.csv")

# Concatenate the dataframes
frames = [ktc_ict_2022, ktc_ict_2020, ktc_ict_2018]
df = pd.concat(frames)

# Calculate the average benchmark score for each region
df_mean = df.groupby(['Region', 'Year of inclusion']).mean().reset_index()

# Create the multiple bar graph
fig = px.bar(df_mean, x="Year of inclusion", y="Total benchmark score", color="Region", barmode="group", 
             title="Average Benchmark Scores by Year and Region")

# Show the graph
fig.show()


C:\Users\kokuo\AppData\Local\Temp\ipykernel_17704\3054604719.py:15: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



In [31]:
import pandas as pd
import plotly.graph_objects as go

# Load the three dataframes
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")
ktc_ict_2020 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2020 KCT ICT.csv")
ktc_ict_2018 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2018 KCT ICT.csv")

# Concatenate the dataframes
frames = [ktc_ict_2022, ktc_ict_2020, ktc_ict_2018]
df = pd.concat(frames)

# Calculate the average benchmark score for each region
df_mean = df.groupby(['Region', 'Year of inclusion']).mean().reset_index()

# Create the multiple bar graph
fig = go.Figure(data=[
    go.Bar(name='2018', x=df_mean[df_mean['Year of inclusion'] == 2018]['Region'], y=df_mean[df_mean['Year of inclusion'] == 2018]['Total benchmark score']),
    go.Bar(name='2020', x=df_mean[df_mean['Year of inclusion'] == 2020]['Region'], y=df_mean[df_mean['Year of inclusion'] == 2020]['Total benchmark score']),
    go.Bar(name='2022', x=df_mean[df_mean['Year of inclusion'] == 2022]['Region'], y=df_mean[df_mean['Year of inclusion'] == 2022]['Total benchmark score'])
])

# Update the layout
fig.update_layout(barmode='group', xaxis_title='Region', yaxis_title='Total benchmark score')

# Show the graph
fig.show()


C:\Users\kokuo\AppData\Local\Temp\ipykernel_17704\2567220824.py:14: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



In [30]:
import pandas as pd
import plotly.graph_objects as go

# Load the three dataframes
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")
ktc_ict_2020 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2020 KCT ICT.csv")
ktc_ict_2018 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2018 KCT ICT.csv")

# Concatenate the dataframes
frames = [ktc_ict_2022, ktc_ict_2020, ktc_ict_2018]
df = pd.concat(frames)

# Calculate the average benchmark score for each region
df_mean = df.groupby(['Region', 'Year of inclusion']).mean().reset_index()

# Define a color dictionary for each region
color_dict = {
    'North America': 'green',
    'Asia': 'red',
    'Europe': 'blue'
}

# Create the multiple bar graph
fig = go.Figure(data=[
    go.Bar(
        name='2018',
        x=df_mean[df_mean['Year of inclusion'] == 2018]['Region'],
        y=df_mean[df_mean['Year of inclusion'] == 2018]['Total benchmark score'],
        marker_color=[color_dict[r] for r in df_mean[df_mean['Year of inclusion'] == 2018]['Region']]
    ),
    go.Bar(
        name='2020',
        x=df_mean[df_mean['Year of inclusion'] == 2020]['Region'],
        y=df_mean[df_mean['Year of inclusion'] == 2020]['Total benchmark score'],
        marker_color=[color_dict[r] for r in df_mean[df_mean['Year of inclusion'] == 2020]['Region']]
    ),
    go.Bar(
        name='2022',
        x=df_mean[df_mean['Year of inclusion'] == 2022]['Region'],
        y=df_mean[df_mean['Year of inclusion'] == 2022]['Total benchmark score'],
        marker_color=[color_dict[r] for r in df_mean[df_mean['Year of inclusion'] == 2022]['Region']]
    )
])

# Update the layout
fig.update_layout(
    barmode='group',
    xaxis_title='Region',
    yaxis_title='Total benchmark score'
)

# Show the graph
fig.show()


C:\Users\kokuo\AppData\Local\Temp\ipykernel_17704\1226192671.py:14: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



In [38]:
import pandas as pd
import plotly.graph_objects as go

# Load the three dataframes
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")
ktc_ict_2022.set_index('Company Name').T

# Create the bar chart
fig = go.Figure(data=[
    go.Bar(name='Theme 1', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 1: Commitment & Governance']),
    go.Bar(name='Theme 2', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 2: Traceability & Risk Assessment']),
    go.Bar(name='Theme 3', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 3: Purchasing Practices']),
    go.Bar(name='Theme 4', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 4: Recruitment']),
    go.Bar(name='Theme 5', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 5: Worker Voice']),
    go.Bar(name='Theme 6', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 6: Monitoring']),
    go.Bar(name='Theme 7', y=ktc_ict_2022.index, x=ktc_ict_2022['Theme 7: Remedy'])
])

# Update the layout
fig.update_layout(barmode='group', xaxis_title='Score', yaxis_title='Theme')

# Show the graph
fig.show()


In [39]:
import pandas as pd
import plotly.graph_objects as go

# Load the three dataframes
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")
ktc_ict_2022.set_index('Company Name', inplace=True)

# Create the bar chart
fig = go.Figure(data=[
    go.Bar(name='Theme 1', x=ktc_ict_2022['Theme 1: Commitment & Governance'], y=ktc_ict_2022.index, orientation='h'),
    go.Bar(name='Theme 2', x=ktc_ict_2022['Theme 2: Traceability & Risk Assessment'], y=ktc_ict_2022.index, orientation='h'),
    go.Bar(name='Theme 3', x=ktc_ict_2022['Theme 3: Purchasing Practices'], y=ktc_ict_2022.index, orientation='h'),
    go.Bar(name='Theme 4', x=ktc_ict_2022['Theme 4: Recruitment'], y=ktc_ict_2022.index, orientation='h'),
    go.Bar(name='Theme 5', x=ktc_ict_2022['Theme 5: Worker Voice'], y=ktc_ict_2022.index, orientation='h'),
    go.Bar(name='Theme 6', x=ktc_ict_2022['Theme 6: Monitoring'], y=ktc_ict_2022.index, orientation='h'),
    go.Bar(name='Theme 7', x=ktc_ict_2022['Theme 7: Remedy'], y=ktc_ict_2022.index, orientation='h')
])

# Update the layout
fig.update_layout(barmode='group', xaxis_title='Theme', yaxis_title='Score')

# Show the graph
fig.show()


In [42]:
import pandas as pd
import plotly.graph_objects as go

# Load the data frame
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")

# Get the relevant columns
data = ktc_ict_2022.loc[:, 'Theme 1: Commitment & Governance':'Theme 7: Remedy']

# Create the bar chart
fig = go.Figure()
for column in data.columns:
    fig.add_trace(go.Bar(x=data[column], y=data.index, name=column))

# Update the layout
fig.update_layout(barmode='group', xaxis_title='Score', yaxis_title='Company')

# Show the graph
fig.show()


In [45]:
import pandas as pd
import plotly.graph_objects as go

# Load the data frame
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")

# Calculate the mean scores
mean_scores = ktc_ict_2022.loc[:, 'Theme 1: Commitment & Governance':'Theme 7: Remedy'].mean()

# Create the bar chart
fig = go.Figure()
fig.add_trace(go.Bar(x=mean_scores, y=mean_scores.index, orientation='h'))

# Update the layout
fig.update_layout(xaxis_title='Mean Score', yaxis_title='Theme')

# Show the graph
fig.show()


In [54]:
import pandas as pd
import plotly.graph_objects as go

# Load the data frame
ktc_ict_2022 = pd.read_csv("C:/Users/kokuo/OneDrive/Study Files/Python/Visualisation Experiments/Know the Chain/KTC ICT Benchmark CSV Data/2022 KCT ICT.csv")

# Calculate the mean scores
mean_scores = ktc_ict_2022.loc[:, 'Theme 1: Commitment & Governance':'Theme 7: Remedy'].mean()

# Create the bar chart
fig = go.Figure()
fig.add_trace(go.Bar(x=mean_scores, y=mean_scores.index, orientation='h'))

# Add scatter plot for individual data points
for region in ktc_ict_2022['Region'].unique():
    region_data = ktc_ict_2022.loc[ktc_ict_2022['Region'] == region, :]
    fig.add_trace(go.Scatter(x=region_data.mean(), y=region_data['Company Name'], mode='markers', name=region))

# Update the layout
fig.update_layout(xaxis_title='Mean Score', yaxis_title='Theme')

# Show the graph
fig.show()


C:\Users\kokuo\AppData\Local\Temp\ipykernel_17704\3766563802.py:17: FutureWarning:

The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.

C:\Users\kokuo\AppData\Local\Temp\ipykernel_17704\3766563802.py:17: FutureWarning:

The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.

C:\Users\kokuo\AppData\Local\Temp\ipykernel_17704\3766563802.py:17: FutureWarning:

The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of n

# **Annex: Data Sorting & Exploration** #

## Basic Data Sorting ##

## Deep Dive Sorting & Exploration ##

# ***Notes*** # 

I have three dataframes in the format below, and i want to present the data on a multiple bar graph, with data from column 'year of inclusion' along the x axis, and 'total benchmark score' along the y axis:

Company Name	Total benchmark score	2022 Rank	Country	Region	Market cap (US$bn)	Year of inclusion
Hewlett Packard Enterprise Co. (HPE)	63.42045455	1	United States	North America	18770.52	2018
Intel Corp.	60.23863636	2	United States	North America	200096.4	2016

I want to use Pandas and plotly to do so